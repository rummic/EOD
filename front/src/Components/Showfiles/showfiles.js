import React, { Component } from "react";
import Sidebar from "../Navbar/sidebar";
import { Redirect, Link } from "react-router-dom";
import { Table } from "react-bootstrap";

import Search from '@material-ui/icons/Search';
import MaterialTable from "material-table";

const token = sessionStorage.getItem("token");
const tableIcons = {
  Search: Search
}
class showfiles extends Component {
  constructor(props) {
    super(props);
    this.state = {
      cases: []
    };
    this.OnChange = this.OnChange.bind(this);
  }

  OnChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  componentDidMount() {
    document.title = 'Lista dokumentów';
    fetch("https://localhost:44388/api/Cases", {
      method: "GET",
      headers: {
        "Content-Type": "aplication/json",
        Authorization: `bearer ${token}`
      }
    }).then(response =>
      response.json().then(responseJSON => {
        console.log(responseJSON);
        this.setState({
          cases: responseJSON.value || []
        });
      })
    );
  }

  render() {
    if (!sessionStorage.getItem("token")) {
      return <Redirect to={"/home"} />;
    }
    return (
      <div className="UsersetBox">
        <Sidebar history={this.props.history} />
        <div className="UsersetBox-content">
          <div className="UsersetBox-form">
            <div className="UsersetBox-form-content">
              <div className="UsersetBox">
                
                  <MaterialTable
        icons = {tableIcons}
          columns={[
            { title: "Tytuł", field: "title" },
            { title: "Dzial", field: "departmentName" },
            {title: "Status", field: "state"},
            {title: "Data wysłania", field: "sendDate"},
            
            {title: "", field:"buttonek"}
            
          ]}
         
           data={this.state.cases.map((item, i) => ({ 
              
           title:  item.title,
           departmentName: item.departmentName,
           state: item.status,
           sendDate: item.sendDate,
           buttonek:<Link to={{pathname:"./filedetails", state: item}}> <button>Ustawienia </button></Link>
           
        
            
})) }

          title="Lista użytkowników"
          />
                </div>
              </div>
            </div>
          </div>
        </div>
    );
  }
}

export default showfiles;
