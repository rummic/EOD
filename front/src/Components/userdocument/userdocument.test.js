import React from "react";
import showfiles from "./userdocument";
import { shallow } from 'enzyme';

describe('showfiles', () => {
    test('Rendering showfiles component', () => {
        const wrapper = shallow(<showfiles />);
        expect(wrapper).toMatchSnapshot();
    });
});  