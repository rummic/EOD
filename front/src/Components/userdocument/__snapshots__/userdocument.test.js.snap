// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`showfiles Rendering showfiles component 1`] = `ShallowWrapper {}`;
